src/qt/transactiontablemodel.cpp

```diff
@@ -309,15 +313,12 @@ class TransactionTablePriv
             // If a status update is needed (blocks came in since last check),
             //  update the status of this transaction from the wallet. Otherwise,
             // simply re-use the cached status.
-            TRY_LOCK(cs_main, lockMain);
-            if (lockMain) {
-                TRY_LOCK(wallet->cs_wallet, lockWallet);
-                if (lockWallet && rec->statusUpdateNeeded()) {
-                    std::map<uint256, CWalletTx>::iterator mi = wallet->mapWallet.find(rec->hash);
-
-                    if (mi != wallet->mapWallet.end()) {
-                        rec->updateStatus(mi->second);
-                    }
+            TRY_LOCK(wallet->cs_wallet, lockWallet);
+            if (lockWallet && rec->statusUpdateNeeded(cur_block_num)) {
+                std::map<uint256, CWalletTx>::iterator mi = wallet->mapWallet.find(rec->hash);
+
+                if (mi != wallet->mapWallet.end()) {
+                    rec->updateStatus(mi->second, cur_block_num);
                 }
```

Comment:
wouldn't it make more sense to check if `rec->statusUpdateNeeded(cur_block_num)` *before* trying to lock `cs_wallet`?

Reply:
yes, it's done in one of the last commits.

Reply:
we should squash all these changes with the newly introduced code.
It would be easier to review... especially for a PR that has 60+ commits.